### ISSUE TITLE
Unable to get shadow master into sync with master on very active cluster.

### ISSUE THREAD

**Issue by 4Dolio** [lizardfs/lizardfs#371]
I just migrated from MooseFS1.6.27 to LizardFS 2.6.0 for our largest cluster. Already running two smaller 2.6.0 clusters which seem to be working fine with my new corosync ocf script. But the recent migration of a much larger and more active cluster has some problems. I can not get the shadow master to properly sync up. even outside of cosync controls it spend far far to long and encounters SIGPROF and SIGVTALRM which I must trap and even then it eventually fails the final sync verification and unloads the filesystem only to retry and fail indefinitely.

The metadata set is about 16G on disk and 50G in memory and we experience 500 to 5000 transactions per second (metadata_version increases). My initial hunch is that perhaps our metadata is moving to quickly for the final sync verification to succeed due to some asynchronicity problem with that process? Any ideas?

I was using the .deb package but have instead compiled our own and the shadow master now behaves slightly differently. This time it outputs these messages, once per metadata version with the final number increasing by one each time:
`wb00007 mfsmaster[7453]: metadata file /var/lib/mfs/metadata.mfs read (174110003 inodes including 24685959 directory inodes and 149377753 file inodes, 39683460 chunks)`
`wb00007 mfsmaster[7453]: running in shadow mode - applying changelogs from /var/lib/mfs`
`wb00007 mfsmaster[7453]: changelog.mfs.1:67298633003: error: 42 (Metadata checksum not matching)`
`mfsmaster[7453]: can't load downloaded metadata and changelogs: can't apply changelog /var/lib/mfs/changelog.mfs.1 (Metadata checksum not matching)`
`mfsmaster[7453]: some changes lost: [67300206869-67300649005], download metadata again`
`mfsmaster[7453]: Waiting for master to produce up-to-date metadata image`
That is 439,136 metadata version changes, presumably during the time it took to download the initial metadata set?

Are there any tunables I can use to compensate for this?

**Comment by 4Dolio** [comment]
Attempting to use the `DISABLE_METADATA_CHECKSUM_VERIFICATION = 1` config setting on the shadow does not change the behaviour, it still ends up saying `synced at version = $something_old` and never actually succeeds...

**Comment by onlyjob** [comment]
First of all if you are using .DEB packages then please make sure you are using official ones from native Debian repository. If you are using Debian _Jessie_ then you will need to enable _jessie-backports_ repository. It is always best to avoid vendor packages whenever possible.

As for shadow master it seems that its data files are corrupted or incomplete. I'd probably just delete its data files and let it re-download metadata from master. Did you try that?

**Comment by 4Dolio** [comment]
Am running Debian 7.9 Wheezy.
I got the packages from packages.lizardfs.com, bad idea?

**Comment by onlyjob** [comment]
> I got the packages from packages.lizardfs.com, bad idea?

Yes, I think so. However there are no official packages for Wheezy (and there are no plans to release 'em unless someone would be willing to sponsor the effort). What you can do is to re-build packages from source or (better) upgrade to Jessie and install from _jessie-backports_. Re-building packages under Wheezy should be relatively easy: put something like `deb-src http://httpredir.debian.org/debian/ jessie-backports main` to `/etc/apt/sources.list` then `apt-get update; apt-get source lizardfs; debuild -us -uc -b`.

> Perhaps I need to compile my own master service on the master node and do a stop/start there? Could be that the .deb master node is the problem?

I don't know, sorry...

**Comment by 4Dolio** [comment]
Looks like I'll compile my own tomorrow...
